modules/drivers/gnss/parser/forsense_parser/forsense_base_parser.cc

```diff
@@ -0,0 +1,184 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *****************************************************************************/
+
+// An parser for decoding binary messages from a NovAtel receiver. The following
+// messages must be
+// logged in order for this parser to work properly.
+//
+#include "modules/drivers/gnss/parser/forsense_parser/forsense_base_parser.h"
+
+#include <cmath>
+#include <iostream>
+#include <limits>
+#include <memory>
+#include <vector>
+
+namespace apollo {
+namespace drivers {
+namespace gnss {
+
+ForsenseBaseParser::ForsenseBaseParser(const config::Config& config) {}
+
+void ForsenseBaseParser::GetMessages(MessageInfoVec* messages) {
+  if (data_ == nullptr) {
+    return;
+  }
+  if (!PrepareMessage()) {
+    return;
+  }
+
+  FillGnssBestpos();
+  FillImu();
+  FillHeading();
+  FillIns();
+  FillInsStat();
+
+  messages->push_back(MessageInfo{MessageType::BEST_GNSS_POS,
+                                  reinterpret_cast<MessagePtr>(&bestpos_)});
+  messages->push_back(
+      MessageInfo{MessageType::IMU, reinterpret_cast<MessagePtr>(&imu_)});
+  messages->push_back(MessageInfo{MessageType::HEADING,
+                                  reinterpret_cast<MessagePtr>(&heading_)});
+  messages->push_back(
+      MessageInfo{MessageType::INS, reinterpret_cast<MessagePtr>(&ins_)});
+  messages->push_back(MessageInfo{MessageType::INS_STAT,
+                                  reinterpret_cast<MessagePtr>(&ins_stat_)});
+}
+
+void ForsenseBaseParser::PrepareMessageStatus(const uint8_t& system_state,
+                                           const uint8_t& satellite_status) {
+  switch (system_state) {
+    case 0:
+      decode_message_.solution_status = SolutionStatus::COLD_START;
+      break;
+    case 1:
+    case 2:
+      decode_message_.solution_status = SolutionStatus::SOL_COMPUTED;
+      break;
+    default:
+      decode_message_.solution_status = SolutionStatus::INSUFFICIENT_OBS;
+  }
+  switch (satellite_status) {
+    case 0:
+      decode_message_.solution_type = SolutionType::NONE;
+      break;
+    case 1:
+    case 6:
+      decode_message_.solution_type = SolutionType::SINGLE;
+      break;
+    case 2:
+    case 7:
+      decode_message_.solution_type = SolutionType::PSRDIFF;
+      break;
+    case 3:
+      decode_message_.solution_type = SolutionType::PROPOGATED;
+      break;
+    case 4:
+      decode_message_.solution_type = SolutionType::INS_RTKFIXED;
+      break;
+    case 5:
+      decode_message_.solution_type = SolutionType::INS_RTKFLOAT;
+      break;
+    case 8:
+      decode_message_.solution_type = SolutionType::NARROW_INT;
+      break;
+    case 9:
+      decode_message_.solution_type = SolutionType::NARROW_FLOAT;
+      break;
+    default:
+      decode_message_.solution_type = SolutionType::NONE;
+  }
+}
+
+void ForsenseBaseParser::FillGnssBestpos() {
+  bestpos_.set_measurement_time(decode_message_.unix_timestamp_sec);
```

Comment:
use gps time，it will be set as unix time in data_parser.cc .

Reply:
change ok